### PR TITLE
Android: Add a button for accessing controller mappings

### PR DESCRIPTION
**Pull request by JosJuice**
The settings GameCube Controller N and Wii Remote N (where N is a number) have two purposes: You can select what controller type you want to use, and also, when you select a controller type (even if you're selecting the one that already is selected), the mapping settings open. This second part is less discoverable than it ideally should be. I'm changing it so that there now is a button for opening the mapping settings instead.

### CODE REVIEW

**Comment by t895** [comment]
I like this change, but could we use a settings cog icon instead? I feel like that fits better for the large set of options you get in the next menu. Then we could use the three dot icon just for the advanced mapping dialog. What do you think?

**Comment by JosJuice** [comment]
> I like this change, but could we use a settings cog icon instead? I feel like that fits better for the large set of options you get in the next menu. Then we could use the three dot icon just for the advanced mapping dialog. What do you think?

Yes, using the settings icon sounds like a good idea. The three dots made it seem just a bit too insignificant.

Undecided on whether we should still go with the three dots for the advanced mapping dialog or if we should make it more consistent, but that decision doesn't have to be made in this PR.

> Edit - I'm also just noticing now, when you have a controller disabled, the button just does nothing. That's not great in its current state. Is there a way you could set that view to `GONE` when disabled?

It would be possible, but it would be annoying to implement cleanly because the logic for which choices do anything is in a different part of the code base. Currently we're jumping through a good number of method calls when the button is pressed. I was kind of pushing the issue aside because I figured setting the controller type to None and then pressing the button would happen pretty rarely, but what do you think?

**Comment by t895** [comment]
> I figured setting the controller type to None and then pressing the button would happen pretty rarely

You're probably right but it doesn't really feel right to just leave that alone

**Comment by JosJuice** [comment]
I'll try to get it working then.

**Comment by JosJuice** [comment]
Done.

File: Source/Android/app/src/main/res/layout/list_item_setting.xml [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
     xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="wrap_content"
     android:background="?android:attr/selectableItemBackground"
     android:clickable="true"
     android:focusable="true"
-    android:minHeight="72dp">
+    android:minHeight="72dp"
```
Comment (t895) [inline Source/Android/app/src/main/res/layout/list_item_setting.xml:11]:
```suggestion
```

We don't need this here given that the layout_height is set to wrap_content (Should also be set in the rtl version)

File: Source/Android/app/src/main/res/layout-ldrtl/list_item_setting.xml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        android:layout_height="wrap_content"
+        android:layout_alignParentStart="true"
+        android:layout_alignStart="@+id/text_setting_name"
+        android:layout_below="@+id/text_setting_name"
+        android:layout_marginBottom="@dimen/spacing_large"
+        android:layout_marginEnd="@dimen/spacing_large"
+        android:layout_marginStart="@dimen/spacing_large"
+        android:layout_marginTop="@dimen/spacing_small"
+        android:textAlignment="viewStart"
+        tools:text="@string/overclock_enable_description" />
```
Comment (t895) [inline Source/Android/app/src/main/res/layout-ldrtl/list_item_setting.xml:41]:
```suggestion
    <TextView
        android:id="@+id/text_setting_name"
        style="@style/TextAppearance.MaterialComponents.Headline5"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_alignParentStart="true"
        android:layout_alignParentTop="true"
        android:layout_marginEnd="@dimen/spacing_large"
        android:layout_marginStart="@dimen/spacing_large"
        android:layout_marginTop="@dimen/spacing_large"
        android:textSize="16sp"
        android:textAlignment="viewStart"
        android:layout_toStartOf="@+id/button_more_settings"
        tools:text="Setting Name" />

    <TextView
        android:id="@+id/text_setting_description"
        android:layout_width="match_parent"
        android:layout_height="wrap_content"
        android:layout_alignParentStart="true"
        android:layout_alignStart="@+id/text_setting_name"
        android:layout_below="@+id/text_setting_name"
        android:layout_marginBottom="@dimen/spacing_large"
        android:layout_marginEnd="@dimen/spacing_large"
        android:layout_marginStart="@dimen/spacing_large"
        android:layout_marginTop="@dimen/spacing_small"
        android:layout_toStartOf="@+id/button_more_settings"
        android:textAlignment="viewStart"
        tools:text="@string/overclock_enable_description" />
```

GitHub won't let me suggest this in the ltr layout file but you should apply it similarly there too.

This view would normally have the text overlap the button when long enough. But conveniently, setting match_parent as the width and ending the text view at the start of the button makes sure we avoid the button as the screen width gets smaller or as the text gets long enough. Setting the view as match_parent has the added benefit of making sure the text view bounds don't get crushed when we set the more settings button as gone.

Once again this also makes me sad that there isn't an option for `nextFocusStart` so we wouldn't even have to make this file...

**Review by t895 — CHANGES_REQUESTED** [review]
After these changes, LGTM

**Comment by JosJuice** [comment]
@t895 Done. But please check that I did it correctly, because GitHub didn't do a nice line-by-line formatting of your second diff for some reason.

**Comment by t895** [comment]
Oh sorry, I messed up a bit. Could you please add 
`android:layout_toStartOf="@+id/button_more_settings"`
to text_setting_description in both layouts?

**Comment by JosJuice** [comment]
Sure thing.

**Review by t895 — APPROVED** [review]

**Review by lioncash — APPROVED** [review]

**Pull request merged**